test/zIntegrationTests.ts

```diff
@@ -1,48 +1,131 @@
-import { helperResetNetwork, helperSwapETHWithOUSD } from "./MainnetHelper";
-import { ethers } from "hardhat";
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { parseEther, formatEther } from "ethers/lib/utils";
 import { expect } from "chai";
 import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
+import { helperResetNetwork, helperSwapETHWithOUSD, defaultBlockNumber, helperSwapETHWith3CRV } from "./MainnetHelper";
+import { createAndFundMetapool, fundMetapool } from "./CurveHelper";
 
-/* Integration tests start here */
+let r: ContractTestContext;
+let owner: SignerWithAddress;
+let user: SignerWithAddress;
+let pretenderOUSDSigner : SignerWithAddress;
+let lvUSD3CRVPoolInstance;
 
-let contractARCHToken;
+const userOUSDPrinciple = 100;
+const initialFundsInPool = 700;
+const initialCoordinatorLvUSDBalance = 10000;
 
-describe("Setting the stage: Getting some OUSD and deploying our contracts", function () {
-    let signer;
-    let r: ContractTestContext;
-    let lvUSD;
+let adminInitial3CRVBalance: number;
 
+function parseEtherNum (num) {
+    return parseEther(num.toString());
+}
+
+function getFloatFromBigNum (bigNumValue) {
+    return parseFloat(formatEther(bigNumValue));
+}
+
+async function printPoolState (poolInstance) {
+    console.log(
+        "Pool has %s coin0/lvUSD and %s coin1/3CRV",
+        getFloatFromBigNum(await poolInstance.balances(0)),
+        getFloatFromBigNum(await poolInstance.balances(1)),
+    );
+}
+
+async function setupEnvForIntegrationTests () {
+    // Reset network before integration tests
+    await helperResetNetwork(defaultBlockNumber);
+
+    // Setup & deploy contracts
+    r = await buildContractTestContext();
+    owner = r.owner;
+    user = r.addr1;
+    pretenderOUSDSigner = r.addr3;
```

Comment:
Can we make this something like `pretendOUSDRebaserSigner`? "Rebaser" for me is what communicates what this is most quickly

Reply:
changed